camel/agents/task_agent.py

```diff
@@ -201,36 +202,43 @@ class TaskCreationAgent(ChatAgent):
             (default: :obj:`None`)
         output_language (str, optional): The language to be output by the
             agent. (default: :obj:`None`)
+        message_window_size (int, optional): The maximum number of previous
+            messages to include in the context window. If `None`, no windowing
+            is performed. (default: :obj:`None`)
     """
 
     def __init__(
         self,
+        role_name: str,
         objective: Union[str, TextPrompt],
         model: Optional[ModelType] = None,
         model_config: Optional[Any] = None,
         output_language: Optional[str] = None,
+        message_window_size: Optional[int] = None,
     ) -> None:
 
         task_creation_prompt = TextPrompt(
-            """Create new tasks with the following objective: {objective}.
-The last completed task has the result:
-{task_result}.
-This result was based on this task: {task}.
-Based on the result, return a list of tasks to be completed in order to meet \
-the objective.
-
-{unsolved_tasks}
-
-Return one task per line in your response.
+            """Create new a task with the following objective: {objective}.
+Never forget you are a Task Creator and I am {role_name}.
```

Comment:
How about letting the `user_role_name` as well?
```suggestion
Never forget you are a Task Creator of {user_role_name} and I am {assisitant_role_name}.
```

Reply:
I feel it should be this way:
Never forget you are a Task Creator of {assisitant_role_name} and I am {user_role_name}.
Task agent is an auxiliary module of the assistant agent to help it with the task creation, planning. I don't think it needs to know user_role_name. In my opinion, it should be
Never forget you are a Task Creator of {assisitant_role_name}.

But, I tried both and didn't find difference in output, probably only nuances that I can't tell.